spec/services/saml_request_validator_spec.rb

```diff
@@ -136,6 +136,93 @@
       end
     end
 
+    context 'unsupported authn context with step up and valid sp and nameID format' do
+      it 'returns a FormResponse with success: true for Comparison=minimum' do
+        Saml::Idp::Constants::PASSWORD_AUTHN_CONTEXT_CLASSREFS.each do |password_context|
+          sp = ServiceProvider.from_issuer('http://localhost:3000')
+          authn_context = [password_context]
+          comparison = 'minimum'
+          name_id_format = Saml::Idp::Constants::NAME_ID_FORMAT_PERSISTENT
+          extra = {
+            authn_context: authn_context,
+            service_provider: sp.issuer,
+            nameid_format: name_id_format,
+          }
+
+          response = SamlRequestValidator.new.call(
+            service_provider: sp,
+            authn_context: authn_context,
+            authn_context_comparison: comparison,
+            nameid_format: name_id_format,
+          )
+
+          expect(response.to_h).to include(
+            success: true,
+            errors: {},
+            **extra,
+          )
+        end
+      end
+
+      it 'returns a FormResponse with success: true for Comparison=better' do
+        Saml::Idp::Constants::PASSWORD_AUTHN_CONTEXT_CLASSREFS.each do |password_context|
+          sp = ServiceProvider.from_issuer('http://localhost:3000')
+          authn_context = [password_context]
+          comparison = 'better'
+          name_id_format = Saml::Idp::Constants::NAME_ID_FORMAT_PERSISTENT
+          extra = {
+            authn_context: authn_context,
+            service_provider: sp.issuer,
+            nameid_format: name_id_format,
+          }
+
+          response = SamlRequestValidator.new.call(
+            service_provider: sp,
+            authn_context: authn_context,
+            authn_context_comparison: comparison,
+            nameid_format: name_id_format,
+          )
+
+          expect(response.to_h).to include(
+            success: true,
+            errors: {},
+            **extra,
+          )
```

Comment:
so IMO we should either be doing `expect().to eq(` and having the full object, or `expect().to include(` and having a partial object

Reply:
@zachmargolis Mind if we defer this to a refactoring ticket assuming that the test actually works as written? I'd love to get this merged and deployed to INT ASAP so that the FSD team can test it today. FWIW I just copied other tests from the file for this so there are a bunch of examples that we'd want to update.

Reply:
Sure, that SGTM

Reply:
Thanks!